lib/dao.js

```diff
@@ -840,6 +840,11 @@ DataAccessObject.replaceOrCreate = function replaceOrCreate(data, options, cb) {
     return this.create(data, options, cb);
   }
 
+  var forceId = this.settings.forceId;
+  if (forceId && id) {
```

Comment:
I don't think if you need to check `id` here; if `id` is `undefined` or `null` it already calls `this.create(...)` please see L840; in other words I think you need to change your condition to:
```
if(Model.settings.forceId && isOriginalQuery) {
  return self.replaceById(id, data, options, cb);
}

```